src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -462,13 +608,16 @@ impl<P: PhysToVirt> PageTableWalker<P> {
             if let Some(frame) = allocator.allocate_frame() {
                 entry.set_frame(
                     frame.frame(),
-                    PageTableFlags::PRESENT | PageTableFlags::WRITABLE,
+                    PageTableFlags::PRESENT | PageTableFlags::WRITABLE | insert_flags,
```

Comment:
I think we can just pass `insert_flags` here. Since we access the page table frame through a separate mapping, the `PRESENT` and `WRITABLE` flags should be only needed if the page is mapped with these flags (or if the user explicitly specified them in `parent_table_flags`).